Formalist/FormNumberFormatter.swift

```diff
@@ -6,28 +6,22 @@
 //  Copyright © 2017 Seed Platform, Inc. All rights reserved.
 //
 
-public enum FormNumberType {
-    case EIN
-    case SSN
-    case creditCard
```

Comment:
I think credit card is common enough to have a special field for. I like the [`DateEncodingStrategy`](https://github.com/apple/swift/blob/master/stdlib/public/SDK/Foundation/JSONEncoder.swift#L40) Apple uses. The custom patterns could be expressed using `case custom((String) -> String) -> Void)`.